src/server/ApiRoutes/Search.js

```diff
@@ -44,12 +45,21 @@ function search(searchKeywords = '', page, sortBy, order, field, filters, cb, er
 
   const aggregationQuery = `/aggregations?${encodedAggregationsQueryString}`;
   const resultsQuery = `?${encodedResultsQueryString}&per_page=50`;
+  const queryObj = { query: { searchKeywords, sortBy, order, field, filters } };
 
   // Need to get both results and aggregations before proceeding.
-  Promise.all([nyplApiClientCall(resultsQuery), nyplApiClientCall(aggregationQuery)])
+  Promise.all([
+    nyplApiClientCall(resultsQuery),
+    nyplApiClientCall(aggregationQuery)])
+    .then(response => ResearchNow.searchAjax(queryObj)
+      .then((drbbResults) => {
+        response.push(drbbResults);
+        return response;
+      })
+      .catch(console.error))
```

Comment:
I would make the DRBB API call part of the parallel call instead of first getting results from Discovery and then making a separate call to DRBB. Try using the `Promise.allSettled` API so you can correctly push DRBB results if any, and continue if there are errors.

(You should also rename `searchAjax` since it's being used as a server call here).

Reply:
Thanks for mentioning `Promise.allSettled`! We had talked about that problem with `Promise.all` before awhile ago. Will check that out.

Reply:
Our current configuration is not up to date enough to have `Promise.allSettled` unfortunately. Any suggestions? I did a little research but could not find anything very helpful